core/src/components/router-outlet/router-outlet.tsx

```diff
@@ -111,7 +111,9 @@ export class RouterOutlet implements ComponentInterface, NavOutlet {
             this.ani.easing('cubic-bezier(1, 0, 0.68, 0.28)');
             newStepValue += getTimeGivenProgression([0, 0], [1, 0], [0.68, 0.28], [1, 1], step)[0];
           } else {
-            newStepValue += getTimeGivenProgression([0, 0], [0.32, 0.72], [0, 1], [1, 1], step)[0];
+            this.ani.easing('linear');
```

Comment:
Can you clarify why you made this change? Swiping back does use a non-linear easing curve

Reply:
Sure! So when playing with the iOS Settings app, the transition easing function on swipe back appears to be linear, or at least much, much closer to linear than the open-a-new-page animation (which is more an ease-out function).

In my testing, if you shorten the duration from 540 to 200 without changing the easing function (from ease-out-ish to linear), the animation "appears" too fast relative to iOS Settings. By changing to linear, similar to what Apple uses for swipe back animation, the swipe back animation finishes faster (which fixes the problem in #27748) but critically doesn't "appear" much faster (because of the linear easing function).

This is all me trying to match what Apple is doing for swipe back - which from my trial an error testing is:

-  linear
- max of 200ms to complete.

Reply:
@liamdebeasi sound good! I agree, on closer inspection swipe back does appear to still be nonlinear. However, the swipe back animation does appear (to me at least) to be much closer to linear than the new page animation (as well as shorter duration).

https://github.com/ionic-team/ionic-framework/assets/2166114/266e9344-8d6e-476e-a73f-d3cd342413d5

Reply:
Looks like keeping the curve + changing the max duration to 300ms (instead of 200ms) achieves a very similar result:

| native | ionic |
| - | - |
| <video src="https://github.com/ionic-team/ionic-framework/assets/2721089/30e37dbe-e711-43a3-96ac-1290a3a7270b"></video> | <video src="https://github.com/ionic-team/ionic-framework/assets/2721089/77161fc6-ba26-4f8d-a74e-58266c60fbd7"></video> |

Reply:
@liamdebeasi, can you try comparing a swipe back with the slowest momentum possible to trigger the swipe? That's when I find the timing function differences most apparent. I am happy to take a look this weekend too and whip up some comparison videos - but I figure you already have your environment setup so :)

Reply:
@liamdebeasi I believe slow swipes is the main problem, because it takes 540ms.

To your point on the slow swipe looking way too fast, that is where the change to the timing function that I originally proposed helps out. By changing the timing function to linear and max time to 200ms, it "appears" slower to the eye (and very similar to iOS, in slow swipe situation), but is actually faster at 200ms.